### ISSUE TITLE
attachment_probability does not work in v0.8

### ISSUE THREAD

**Issue by ma-ji** [diging/tethne#154]
Hi, I found the use of attachment_probability in v0.8 is not consistent with former versions, and I cannot work this out with v0.8. Is this a fault of mine or discrepancy between versions?

This is a great package and I love it! However, I feel there should be some plans of developing in a higher level, so that different versions can keep consistent and share the same logic. I would love to contribute, but I'm not a CS guy. I would love to know more about how we can help.

Best,
Ji

**Comment by erickpeirson** [comment]
@ma47 Thanks for the report, and for using Tethne! Yes, there have been quite a few dramatic changes over the last several "minor" versions. Early versions had a sprawling and inefficient code-base, so there has been a lot of refactoring with a focus on core functionality. This will be [bug/TETHNE-128](https://diging.atlassian.net/browse/TETHNE-128).

Honestly, the best way to help keep this project on track is by continuing to use the package as it is developed, and submit issues just like this one. A few other projects have taken precedence over the last year, but given that there has been a rather sudden burst of activity over the last couple of months I'm hoping to devote more time to this project.

The JIRA project is here: https://diging.atlassian.net/projects/TETHNE/ . If you create an account (click Log In in the upper right, follow links), you should be able to submit bug reports directly and keep closer tabs on development.

**Comment by ma-ji** [comment]
@erickpeirson Thank you for your timely reply! I'm fully confident that this package will be much more popular. I was using CiteSpace, but applications like this are pretty limited in manipulating the data. Lots of "quick publications" have been made because of these tools, but they add little theoretical contributions to the discipline. I would like to use packages like tethne, by which we can ask more important questions, but not generate publications according to the "cook book." You have my full support! And I definitely like to introduce tethne to more people, if they really want to do research.

**Comment by ma-ji** [comment]
One more suggestion: you can write a paper to introduce this package and get published somewhere, therefore, we can cite your paper. This will boost your citation, and you really deserve this!

Like this one: http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0085777

**Comment by erickpeirson** [comment]
@ma47 Thank you so much for your support! Yes, you're right about writing a paper, this is something that I have been thinking about -- hopefully this summer...

**Comment by erickpeirson** [comment]
@ma47 I have uploaded a [pre-release "dev" build](https://pypi.python.org/pypi/tethne/0.8.1.dev2) to PyPI with a refactored `attachment_probability`. It would be great if you could try it out and let me know how it goes. 

To install via pip, use the `--pre` flag, e.g.:

``` shell
pip install -U tethne --pre
```

This tells pip to download the latest pre-release version (otherwise it will just download 0.8).

Example usage:

``` python
>>> from tethne.readers.wos import read
>>> corpus = read("/path/to/my/data")
>>> from tethne import coauthors, GraphCollection
>>> collection = GraphCollection(corpus, coauthors)
>>> from tethne.analyze.collection import attachment_probability
>>> probs = attachment_probability(collection)
```

**Comment by ma-ji** [comment]
Sure! will try this out today, thank you!

**Comment by erickpeirson** [comment]
@ma47 Excellent thanks. Also, I created a Google Q/A group for Tethne users: https://groups.google.com/forum/#!forum/tethne-users

**Comment by erickpeirson** [comment]
Fixed in 3164186, this will be released with v0.8.1-beta.
